src/Controller/HomeController.php

```diff
@@ -2,17 +2,38 @@
 
 namespace App\Controller;
 
+use App\Repository\SnowFigureRepository;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
 
 class HomeController extends AbstractController
 {
     #[Route('/', name: 'home')]
-    public function index(): Response
+    public function index(SnowFigureRepository $snowFigureRepository): Response
     {
+        //list of published figures, by decreasing id
+        $figuresPublished = $snowFigureRepository->findBy(['publish' => '1'], ['id' => 'DESC']);
+        $figuresPublishedHome = $snowFigureRepository->findBy(['publish' => '1'], ['id' => 'DESC'], 5, 0);
+
+        //total number of figures published
+        $nbFiguresPublished = count($figuresPublished);
+
         return $this->render('home/index.html.twig', [
             'controller_name' => 'HomeController',
+            'figures' => $figuresPublishedHome,
+            'nbFigures' => $nbFiguresPublished,
+        ]);
+    }
+
+    #[Route('/blockFigures', name: 'blockFigures', methods: 'GET')]
+    public function blockFigures(SnowFigureRepository $snowFigureRepository): Response
+    {
+        $numFigure = $_GET['numFigure'];
```

Comment:
Avec Symfony tu ne dois pas passer par ta variable globale $_GET. Regarde la documentation sur les controller symfony et plus particulièrement cette [partie ](https://symfony.com/doc/current/controller.html#the-request-object-as-a-controller-argument).

Reply:
Remplacé par :

HomeControlleur.php
$numFigure = $request->query->get('numFigure');
$figures = $snowFigureManager->getPublishedNumFigure(5, $numFigure);

SnowFigureManager.php : 
$figures = $this->snowFigureRepository->findBy(['publish' => '1'], ['id' => 'DESC'], $max, $numFigure);